### PR TITLE
removed ISharedAccessSignatureCredential and StringFormatExtensions

### PR DESCRIPTION
**Pull request by tmahmood-microsoft**
## Checklist
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/main/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `main` branch.

### CODE REVIEW

File: iothub/service/src/Authentication/AuthenticationMethodFactory.cs [hunk 1]
```diff
@@ -32,7 +32,7 @@ internal static IAuthenticationMethod GetAuthenticationMethod(IotHubConnectionSt
                 return new ServiceAuthenticationWithSharedAccessPolicyKey(iotHubConnectionStringBuilder.SharedAccessKeyName, iotHubConnectionStringBuilder.SharedAccessKey);
             }
 
-            throw new InvalidOperationException("Unsupported Authentication Method {0}".FormatInvariant(iotHubConnectionStringBuilder));
+            throw new InvalidOperationException($"Unsupported Authentication Method {iotHubConnectionStringBuilder}.");
```
Comment (drwill-ms) [inline iothub/service/src/Authentication/AuthenticationMethodFactory.cs:35]:
```suggestion
            throw new InvalidOperationException($"Unsupported authentication method {iotHubConnectionStringBuilder}.");
```

File: iothub/service/src/Exceptions/IotHubSuspendedException.cs [hunk 2]
```diff
@@ -22,7 +22,7 @@ public class IotHubSuspendedException : IotHubException
         /// </summary>
         /// <param name="iotHubName">The name of the IoT hub that has been suspended.</param>
         public IotHubSuspendedException(string iotHubName)
-            : base("IoT hub {0} is suspended".FormatInvariant(iotHubName))
+            : base($"IoT hub {iotHubName} is suspended")
```
Comment (drwill-ms) [inline iothub/service/src/Exceptions/IotHubSuspendedException.cs:25]:
```suggestion
            : base($"IoT hub {iotHubName} is suspended.")
```

File: iothub/service/tests/TestAssert.cs [hunk 3]
```diff
@@ -24,20 +24,18 @@ public static class TestAssert
...
             catch (Exception ex)
             {
-                throw new AssertFailedException(
-                    "{0}. Expected:<{1}> Actual<{2}>".FormatInvariant(errorMessage, typeof(TException).ToString(), ex.GetType().ToString()), 
-                    ex); 
+                throw new AssertFailedException($"{errorMessage}. Expected:{typeof(TException).ToString()} Actual:{ex.GetType().ToString()}", ex);
             }
 
-            throw new AssertFailedException("{0}. Expected {1} exception but no exception is thrown".FormatInvariant(errorMessage, typeof(TException).ToString()));
+            throw new AssertFailedException($"{errorMessage}. Expected {typeof(TException).ToString()} exception but no exception is thrown.");
```
Comment (drwill-ms) [inline iothub/service/tests/TestAssert.cs:30]: ToString()is implied by the runtime.

```suggestion
            throw new AssertFailedException($"{errorMessage}. Expected {typeof(TException)} exception but no exception is thrown.");
```

File: iothub/service/tests/TestAssert.cs [hunk 4]
```diff
@@ -51,11 +49,10 @@ public static class TestAssert
             catch (Exception ex)
             {
                 throw new AssertFailedException(
-                    "{0}. Expected:<{1}> Actual<{2}>".FormatInvariant(errorMessage, typeof(TException).ToString(), ex.GetType().ToString()),
-                    ex);
+                    $"{errorMessage}. Expected:{typeof(TException).ToString()} Actual:{ex.GetType().ToString()}", ex);
             }
 
-            throw new AssertFailedException("{0}. Expected {1} exception but no exception is thrown".FormatInvariant(errorMessage, typeof(TException).ToString()));
+            throw new AssertFailedException($"{errorMessage}. Expected {typeof(TException).ToString()} exception but no exception is thrown");
```
Comment (drwill-ms) [inline iothub/service/tests/TestAssert.cs:55]:
```suggestion
            throw new AssertFailedException($"{errorMessage}. Expected {typeof(TException)} exception but no exception is thrown");
```

**Review by drwill-ms — APPROVED** [review]

File: iothub/service/src/Registry/ModulesClient.cs [hunk 5]
```diff
@@ -21,8 +21,6 @@ public class ModulesClient
         private readonly HttpClient _httpClient;
         private readonly HttpRequestMessageFactory _httpRequestMessageFactory;
 
-        private const string ModulesRequestUriFormat = "/devices/{0}/modules/{1}";
```
Comment (timtay-microsoft) [inline iothub/service/src/Registry/ModulesClient.cs:24]: I'd like to keep these strings as constants. If we have code like 

```csharp
return new Uri($"/devices/{deviceId}/modules/{moduleId}", UriKind.Relative);
```

It actually allocates a new string on the heap each time. By keeping it as a constant, we just reference the same point on the heap repeatedly. So what you should have below is something like

```csharp
string.Format(CultureInfo.InvariantCulture, ModulesRequestUriFormat, deviceId, moduleId);
```

Comment (tmahmood-microsoft) [reply iothub/service/src/Registry/ModulesClient.cs:24]: added the changes

**Review by timtay-microsoft — APPROVED** [review]

**Pull request merged**